src/main.rs

```diff
@@ -93,6 +93,83 @@ impl ExtrinsicOpts {
     }
 }
 
+#[derive(Clone, Debug, StructOpt)]
+pub struct OptimizationFlags {
+    // number of optimization passes
+    #[structopt(long = "optimization-passes", default_value = "3")]
+    optimization_passes: String,
```

Comment:
I think this function should also contain 
```
    // How much to focus on shrinking the code size
    #[structopt(long = "shrink-level", default_value = "…")]
    shrink_level: String,
```

You seem to derive the `shrink-level` from the `optimization-passes` currently, but in `wasm-opt` it' an independent cli flag.